aspnetcore/host-and-deploy/health-checks.md

```diff
@@ -0,0 +1,614 @@
+---
+title: Health checks in ASP.NET Core
+author: guardrex
+description: Learn how to set up health checks for ASP.NET Core infrastructure, such as apps and databases.
+monikerRange: '>= aspnetcore-2.2'
+ms.author: riande
+ms.custom: mvc
+ms.date: 12/03/2018
+uid: host-and-deploy/health-checks
+---
+# Health checks in ASP.NET Core
+
+By [Luke Latham](https://github.com/guardrex) and [Glenn Condron](https://github.com/glennc)
+
+ASP.NET Core offers Health Check Middleware and libraries for reporting the health of app infrastructure components.
+
+Health checks are exposed by an app as HTTP endpoints. Health check endpoints can be configured for a variety of real-time monitoring scenarios:
+
+* Health probes can be used by container orchestrators and load balancers to check an app's status. For example, a container orchestrator may respond to a failing health check by halting a rolling deployment or restarting a container. A load balancer might react to an unhealthy app by routing traffic away from the failing instance to a healthy instance.
+* Use of memory, disk, and other physical server resources can be monitored for healthy status.
+* Health checks can test an app's dependencies, such as databases and external service endpoints, to confirm availability and normal functioning.
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/host-and-deploy/health-checks/samples) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+The sample app includes examples of the scenarios described in this topic. To run the sample app for a given scenario, use the [dotnet run](/dotnet/core/tools/dotnet-run) command from the project's folder in a command shell. See the sample app's *README.md* file and the scenario descriptions in this topic for details on how to use the sample app.
+
+## Prerequisites
+
+Health checks are usually used with an external monitoring service or container orchestrator to check the status of an app. Before building a production health check app, decide on which monitoring system to use. The monitoring system dictates what types of health checks to create and how to configure their endpoints.
```

Comment:
> building a production health check app

This could be misleading... I'd rather see something here like "before adding health checks to your app..."